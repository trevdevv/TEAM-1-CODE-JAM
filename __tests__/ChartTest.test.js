import React from "react";
import "babel-polyfill";
import renderer from "react-test-renderer";

// Import chart component
import Charts from "../src/components/Charts";

// Expected Chart Data on Render
const ChartData = {
  labels: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"],
  datasets: [
    {
      type: "bar",
      label: "My Contributions",
      data: [10, 20, 5, 8, 10],
      backgroundColor: "rgba(11, 113, 126, 0.7)",
    },
    {
      type: "bar",
      label: "Total Contributions",
      data: [30, 40, 50, 60, 20],
      backgroundColor: [
        "rgba(255,99,132,0.6)",
        "rgba(54,162,235,0.6)",
        "rgba(255,206,86,0.6)",
        "rgba(75,192,192,0.6)",
        "rgba(153,102,255,0.6)",
      ],
    },
  ],
};

// Check charts are rendering correctly
test("Charts are set up correctly", () => {
  const component = renderer.create(<Charts />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Test that the props are rendered correctly
  const { id, chartData, className } = tree.props;

  expect(id).toBe("chart");
  expect(chartData).toStrictEqual(ChartData);
  expect(className).toBe("chart");
});
