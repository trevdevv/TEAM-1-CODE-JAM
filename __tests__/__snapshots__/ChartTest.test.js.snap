// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts are set up correctly 1`] = `
<div
  chartData={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "rgba(11, 113, 126, 0.7)",
          "data": Array [
            10,
            20,
            5,
            8,
            10,
          ],
          "label": "My Contributions",
          "type": "bar",
        },
        Object {
          "backgroundColor": Array [
            "rgba(255,99,132,0.6)",
            "rgba(54,162,235,0.6)",
            "rgba(255,206,86,0.6)",
            "rgba(75,192,192,0.6)",
            "rgba(153,102,255,0.6)",
          ],
          "data": Array [
            30,
            40,
            50,
            60,
            20,
          ],
          "label": "Total Contributions",
          "type": "bar",
        },
      ],
      "labels": Array [
        "Monday",
        "Tuesday",
        "Wednesday",
        "Thursday",
        "Friday",
      ],
    }
  }
  className="chart"
  id="chart"
/>
`;
